### DISCUSSION TITLE
onclick window change image src

### DISCUSSION THREAD

**Discussion by Samuroid** [matteobruni/tsparticles#1725] [Q&A] [answered]
Hi

I'm trying to change the particle image to another image when the user clicks the window.
I've tried to have a go and looking in the documentation but can't figure out how to get access to that setting from my particles instance.

Any pointers?

Many thanks

**Comment by matteobruni** [comment, marked as answer]
Hi @Samuroid,

Changing images like this is not actually possible, you need to change options and refresh the instance.

The main issue with changing images is that they need to be loaded inside the canvas and this operation could fail or could take some time client side resulting in an unexpected canvas state.

I was thinking about a `bubble` hover/click mode where you can change the image hovering or clicking but it's not ready yet.

**Reply by Samuroid** [reply]
What would be the proper way to unload the particle object?
Currently it's a const so I have changed it to a var and redefined it with a new particle.load object.

Yess the onclick image change is something I'm looking to implement exactly!

**Reply by matteobruni** [reply]
I thought about themes and it works nice as you can see in this sample: https://codepen.io/matteobruni/pen/VwbYeYy

I have done some workarounds because I also found a bug but you are ok with this configuration since the workaround is stable there. I'm planning the `1.30.2` version, I try to fix that there.

**Reply by Samuroid** [reply]
I see that is a good idea with the theme options!
That works nicely, I event changed the event listening to the window and it works well so far.

I also found out I can change the image using this line and then refreshing the particles to update it:
`window.particles.options.particles.shape.image["src"] = "path-to-new-img.png"`
`window.particles.refresh()`

**Reply by Samuroid** [reply]
I know it's a long shot, but has anything been implemented to change one particle image/shape when clicking on it?

**Reply by matteobruni** [reply]
> I know it's a long shot, but has anything been implemented to change one particle image/shape when clicking on it?

Not yet, I'm working on a 1.30.2 release and that fix will be part of it too, but there are other things to fix

**Reply by matteobruni** [reply]
Found the bug and solved, you can follow the PR when it will be merged and published here: https://github.com/matteobruni/tsparticles/pull/1726

**Reply by Samuroid** [reply]
Fantastic matt! i look forward to plugin it into my site :)

**Reply by matteobruni** [reply]
I have updated the CodePen sample above with `1.30.2` so you can see the right way of working with themes without the previous workaround
